dev/PushNotifications/PushNotificationManager.cpp

```diff
@@ -111,10 +112,18 @@ namespace winrt::Microsoft::Windows::PushNotifications::implementation
     }
     CATCH_RETURN()
 
+    bool PushNotificationManager::IsSupported()
+    {
+        // Only scenarios that use the Background Infrastructure component of the OS support Push in the SelfContained case
+        static bool isSupported{ !WindowsAppRuntime::SelfContained::IsSelfContained() || PushNotificationHelpers::IsPackagedAppScenario() };
+        return isSupported;
+       
+    }
+
     winrt::Microsoft::Windows::PushNotifications::PushNotificationManager PushNotificationManager::Default()
     {
         THROW_HR_IF(E_NOTIMPL, !::Microsoft::Windows::PushNotifications::Feature_PushNotifications::IsEnabled());
-
+        THROW_HR_IF_MSG(E_FAIL, !IsSupported(), "Push APIs are not supported for this Application Type!");
```

Comment:
The `IsSupported` pattern for WinRT APIs does *not* require the caller check IsSupported before calling. If the thing isn't supported, then future uses of it are "inert" - meaning they do nothing, and provide a reasonable state as a result.  So for instance, calling Register or Unregister[All] has no side-effect, the app never gets an incoming notification, and any requests to open a channel get back the "oops, couldn't do that" state.  But _not_ an exception here.

Reply:
Thanks for pointing this out. I have removed the THROW here and gated the remaining APIs in PushManager to have no side effects for the !IsSupported() case.